Cleanup, additional linting, and build badge

jsf9k opened the pull request:
@dav3r, @felddy, and @KyleEvers, please have a look at this PR.  I've been toying with the idea of running linters in TravisCI on every file I can in each repo.  In this repo I added a Markdown linter, and I wanted to know what you guys thought of doing this in all our repos.  I'm for it, if only to ensure that we standardize.  In cases where you really need to, you can always forcibly disable certain warnings.  In this case, for example, I took the particularly heavy-handed step of  not running the Markdown linter on [LICENSE.md](/cisagov/ansible-role-banner/blob/develop/LICENSE.md) at all, since that file needs to match the license text exactlyin order got GitHub's automatic recognition fu to work.  Other than that, I only had to make a handful of other changes in order to make the linter happy.

jsf9k committed:
Add build badge

jsf9k committed:
Add markdown linter to Travis CI config

jsf9k committed:
Fix typo in email address and make mdl happy

jsf9k committed:
Stop running mdl on LICENSE.md

LICENSE.md has a lot of issues (according to mdl) but it must have the
exact form it takes in order to be automatically recognized by GitHub.

jsf9k committed:
Remove trailing punctuation in header

jsf9k committed:
Cosmetic change

jsf9k committed:
Make linters run against anything not explicitly disallowed

The way the code was written before, the linters were running against
only specific files.  If a new file was added, and .travis.yml wasn't
updated, then the linter would not have run against the new file.
This change remedies that.

dav3r commented on README.md:
```diff
@@ -41,4 +43,4 @@ with this waiver of copyright interest.
 ## Author Information ##
 
-Mark Feldhousen <make.feldhousen@beta.dhs.gov>
+Mark Feldhousen <mark.feldhousen@trio.dhs.gov>
```
LOL - "make.feldhousen" 😃

felddy replied:
I made three Feldhousen.

jsf9k replied:
:baby: :baby: :baby:

dav3r reviewed (approved):
I'm on board with trying to lint all the things- your plan makes sense to me.  We can always change course later if we encounter any problems with this approach.

felddy commented:
I am for linting all the things.  I'd rather a cold lump of metal attack your ego than doing it myself.

jsf9k commented:
Based on @dav3r's and @felddy's comments I'll go full Cajun on this repo and blacken it.

![Blackened Redfish](http://louisiana.kitchenandculture.com/sites/default/files/styles/large/public/recipes/Louisiana%20Kitchen%20%26amp%3B%20Culture/reggios-blackened-redfish.jpg?itok=siCJr0I-)

jsf9k committed:
Add pre-commit config and pip requirements files for dev and test

Also update TravisCI config to use the testing requirements file.

jsf9k committed:
Use pre-commit to run linters in TravisCI

jsf9k committed:
Add mdl to pre-commit hooks

Exclude the LICENSE.md file, since it has a few issues but needs to
match the license text exactly for GitHub's autorecognition foo to
work.

jsf9k committed:
Add flake8 config that matches black's default line length behavior

See https://github.com/ambv/black#line-length for more details.

jsf9k committed:
Remove the ruby build

pre-commit now handles the markdown linting, so it is not needed

jsf9k committed:
Add shellcheck linter to pre-commit config

jsf9k committed:
Add YAML header line to .yamllint

jsf9k committed:
Blacken test_default.py

jsf9k committed:
Add pre-commit hooks for terraform

jsf9k committed:
Add ansible-lint hook and a bunch of pre-commit-hooks hooks

jsf9k committed:
Add hooks to sort requirements.txt, check XML syntax, and check for
merge conflict strings

jsf9k committed:
Allow the AWS credentials hook to pass if no creds are found

There are no creds in the TravisCI build, so the hook was failing
there.

jsf9k committed:
Add prettier (JS linter) hook and run autoupdate

jsf9k committed:
Add comments to .flake8 as requested by @felddy and @dav3r

jsf9k committed:
Add .prettierignore

We don't need to lint markdown or YAML files since they are already
being linted by mdl and yamllint, respectively.

jsf9k committed:
Add a bunch of information about contributing to CONTRIBUTING.md

jsf9k committed:
Add a blurb in README that points the interested to CONTRIBUTING.md

jsf9k committed:
Fix broken links

jsf9k committed:
Fix another broken link

jsf9k committed:
Add a lot more content about setting up pre-commit

jsf9k committed:
Fix a typo

jsf9k commented:
@cisagov/team-ois, please have a look at the additions I made to `CONTRIBUTING.md` and let me know what you think.  Once that is good I will use this as a basis for the bootstrap repo we talked about.

jsf9k committed:
Merge branch 'develop' into improvement/cleanup_and_build_badge
